### PR TITLE
Add from_json method to classes

### PR DESCRIPTION
**Pull request by dstansby**
Adds a abstract base class for objects that can go to/from JSON. I only added the `to_json` so far.

Also adds a basic test against the example metadata given in the v0.4 spec.

### CODE REVIEW

File: src/ome_zarr_models/v04/image.py [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
 from dataclasses import dataclass
-from typing import Any, Literal
-
-from ome_zarr_models.zarr_models.v2 import Group
+from typing import Any, Literal, Self
 
 AxisType = Literal["time", "space", "channel"]
 
 
+class ToFromJSON(ABC):
```
Comment (d-v-b) [inline src/ome_zarr_models/v04/image.py:9]: name suggestion: `JSONable`

File: src/ome_zarr_models/v04/image.py [hunk 2]
```diff
@@ -67,19 +85,53 @@ class TranslationTransform:
...
+    https://ngff.openmicroscopy.org/0.4/index.html#multiscale-md
+    """
+
     path: str
     coordinateTransformations: (
         tuple[ScaleTransform] | tuple[ScaleTransform, TranslationTransform | str]
     )
 
+    @classmethod
+    def _from_json(cls, json) -> Self:
```
Comment (d-v-b) [inline src/ome_zarr_models/v04/image.py:116]: This variable name might cause name collisions if we do `import json`

File: src/ome_zarr_models/v04/image.py [hunk 3]
```diff
@@ -1,15 +1,22 @@
...
 
 AxisType = Literal["time", "space", "channel"]
 
 
+class ToFromJSON(ABC):
+    """A class that can serialise to and from JSON."""
+
+    @classmethod
+    @abstractmethod
+    def _from_json(cls, json: dict) -> Self: ...
```
[the remark this answers is not in the document]
Comment (dstansby) [reply src/ome_zarr_models/v04/image.py:14]: Depends what we want to external API to be? I would say lets make input/output like this private to start, then when we write the tutorial we can decide what we should make public?

File: src/ome_zarr_models/v04/image.py (hunk 1 above)
Comment (dstansby) [reply src/ome_zarr_models/v04/image.py:9]: Ooh I like

File: src/ome_zarr_models/v04/image.py (hunk 2 above)
Comment (dstansby) [reply src/ome_zarr_models/v04/image.py:137]: ooh yes, I will change

**Pull request merged**